Warehouse.Client/warehouse-client/src/app/login/login.component.ts

```diff
@@ -0,0 +1,49 @@
+import {Component, EventEmitter, OnInit, Output} from '@angular/core';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
+import {ActivatedRoute, Router} from '@angular/router';
+import {MyErrorStateMatcher} from '../errors/myErrorStateMatcher';
+import {User} from '../models/user';
+import { AuthService } from '../services/auth/auth.service';
+
+@Component({
+  selector: 'login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+  matcher = new MyErrorStateMatcher();
+  private returnUrl!: string;
+
+  @Output() loginedUser = new EventEmitter<User>();
+
+  formControl = new FormGroup({
+    userName: new FormControl('', [Validators.required, Validators.minLength(5)],),
+    password: new FormControl('', [Validators.required, Validators.minLength(5)])
+  })
+
+  constructor(private route: ActivatedRoute, private router: Router, private authService: AuthService) {
+  }
+
+  async login() {
+    let login = this.formControl.value;
+    let loginResponse: any;
+    let response: any = await this.authService.login(login).toPromise();
+    this.formControl.get('password')?.reset();
+
+    if (response.error) {
+      let controlName = Object.keys(response.error.Errors)[0];
+      this.formControl.get(controlName)?.setErrors({serverError: true});
+      return;
+    }
+
+    localStorage.setItem('jwtToken', response.jwtToken);
```

Comment:
It's better to move this piece of log in logic to a service, because componenta (like controllers in ASP .NET) don't usually do any logic, just manipulate the data